Automattic/jetpack
# Block bundling: introduce single variation constant

Fixes #28227

## Proposed changes:

Instead of having 2 separate constants to manage what block bundle is loaded on a site, let's only use one, `JETPACK_BLOCKS_VARIATION`. It makes it easier to manage and overwrite.

I also sunset 2 existing filters in favor of the existing `jetpack_blocks_variation` filter which works the same way as our new constant.

> **Note**
> Once this gets merged, I'll p2 the change so that everyone contributing to Jetpack can start using the new constant in their dev environments. I've also opened https://github.com/Automattic/companion/pull/61 and 1229-gh-wpcomsh to bring the new constant to our existing environment.

> **Note**
> Since the constants are used by many people, I opted not to display a deprecation notice for those constants just yet. I think it would be noisy. Happy to hear other thoughts on that though.

### Other information:


## Jetpack product discussion

* N/A

## Does this pull request change what data or activity we track or use?

* No

## Testing instructions:

There are different things one can test here:

* Start with a brand new JN site.
* Go to Settings > Jetpack constants.
* Toggle the experimental or beta checkboxes.
* Go to Posts > Add New and check the network tab; you should see a matching `editor-beta.js` or `editor-experimental.js` being loaded.
* Now go back to the Jetpack Constants settings page, and disable the checkbox you checked.
* In a functionality plugin or in your theme's `functions.php` file, add the following: `add_filter( 'jetpack_load_beta_blocks', '__return_true' );`
* Go back to Posts > Add New; the Beta bundle should be loaded.
* Now check your logs at `wp-content/debug.log`; you should see a deprecation warning.
* Remove that filter you added earlier.
* Edit your site's `wp-config.php` and add the following: `define( 'JETPACK_EXPERIMENTAL_BLOCKS', 'true' );`
* Back in the post editor, you should see the experimental bundle
* Edit your site's `wp-config.php` again, and add the following instead: `define( 'JETPACK_BLOCKS_VARIATION', 'beta' );`
* Back in the post editor, you should see the beta bundle loaded.
* Go to Settings > Writing
* You should see the checkbox to enable writing prompts towards the top of the page.
* Now install the Jetpack VideoPress plugin
* In the post editor, you should see the Beta VideoPress block.
